framework-plugins/lisk-framework-monitor-plugin/src/monitor_plugin.ts

```diff
@@ -148,10 +150,21 @@ export class MonitorPlugin extends BasePlugin {
 			const { event, data } = eventData as { event: string; data: unknown };
 
 			if (event === 'postTransactionsAnnouncement') {
+				const errors = validator.validate(
+					transactionAnnouncementSchema,
+					data as Record<string, unknown>,
+				);
+				if (errors.length > 0) {
+					return;
+				}
 				this._handlePostTransactionAnnounce(data as { transactionIds: string[] });
 			}
 
 			if (event === 'postBlock') {
+				const errors = validator.validate(postBlockEventSchema, data as Record<string, unknown>);
+				if (errors.length > 0) {
+					return;
```

Comment:
should we add a debug log here

Reply:
Anyone can spam with random data and logging all can be IO consumption. But for debug mode it would be useful.

Reply:
In this case, the framework will log and ban the peer, so i think it should be sufficient